### DISCUSSION TITLE
Should we deprecate the sync command?

### DISCUSSION THREAD

**Discussion by volllly** [volllly/rotz#334] [Polls]
The `sync` command basically is a wrapper around git that stages all changes and then commits does a pull and then and pushes the changes.

Currently there are some cases where it fails and making it stable would require a lot of work.
I don't actually use it most of the time and just do the git actions manually (or sometimes with vscode).

> Related https://github.com/volllly/rotz/issues/329#issuecomment-2049109361

**Comment by t-mart** [comment]
I'm the linked comment, but my only argument to keep `sync` would be that it allows users to *only* interact with rotz.

In other words, if you consider rotz an "interface", and git a dependency of that interface, then it kind of sucks to have rotz "leak" git to users. Abstraction is a great thing. You only really want to be working with one interface. Therefore, keeping `sync` might be a good thing.

But, git is hairy: it's so configurable and supports many workflows. It's hard to wrangle and generalize. Ultimately, I have trouble imagining any software supporting all that:  managing working tree, staged files, commits, branches, remotes, remote branches, pushes, etc. This would be a large developer burden.

Additionally, and this might just be me speaking, I feel more comfortable with git than another software interacting with it on my behalf.

For this reasons, I'd say: get rid of `sync` and document that repo management should be done *by the user* with git.

If we're considering getting rid of `sync`, you could argue that we look the same at `clone`: another subcommand that wraps git functions. However, I like `clone` because:

- it sets up the `~/.dotfiles` path correctly
- cloning ~is~ seems much simpler than what sync would do

So, again, I'd say that `sync` is something that rotz developers and users do not want (at this time?).

**Comment by Glitchy-Sheep** [comment]
Bro, I don’t even know exactly where I should say this, but THANK YOU SO MUCH for the tool. I absolutely LOVE IT, and I MEAN it like never before. This tool has saved me SO MUCH time, you can’t even imagine. Setting up OBS, editors, shells, prompts, git, and all that stuff—now it literally works with just TWO COMMANDS. Man, I can’t even explain how much more comfortable my life is now. :DD

I just want you to know it.
